lms/static/scripts/frontend_apps/components/StatusChip.tsx

```diff
@@ -0,0 +1,35 @@
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+
+export type StatusChipProps = {
+  /**
+   * Determines the aspect of the status chip, being "top" the one representing
+   * the most successful status, and "bottom" the one representing a total
+   * failure.
+   */
+  variant: 'success' | 'semi-success' | 'warning' | 'error' | 'fatal';
```

Comment:
I'm trying to come up with a better name than `semi-success`. Or perhaps use `success` for that one and think on something which is "above" success.